src/lib/io/sol-gpio-impl-riot.c

```diff
@@ -59,8 +63,26 @@ static void
 gpio_process_cb(void *data)
 {
     struct sol_gpio *gpio = data;
+    bool val;
 
-    gpio->irq.cb((void *)gpio->irq.data, gpio);
+    val = sol_gpio_read(gpio);
+    gpio->irq.cb((void *)gpio->irq.data, gpio, val);
```

Comment:
On Fri, 13 Nov 2015 14:16:42 -0800, Gustavo Sverzut Barbieri wrote:
> >  
> > -    gpio->irq.cb((void *)gpio->irq.data, gpio);
> > +    val = sol_gpio_read(gpio);
> > +    gpio->irq.cb((void *)gpio->irq.data, gpio, val);
> 
> same question as for as Contiki, can we read it in the ISR and queue the value somehow?

Technically yes, but then we'll have RIOT behaving differently than
Linux and Contiki. On those we'll only report the last value available
at the time the user function is called.

On Riot we can implement a queue for the ISR to store the values and
then call the user function repeatedly for each value, but I'd rather
keep it consistent with the others. On that note, it should avoid
putting a new message for each ISR, and instead mark some pending_isr
flag that the main loop will clear when it finally gets to it, but that
should probably be done regardless of whether we queue the values or
not.

Opinions?